### ISSUE TITLE
[4.0] [PHP 7.4.8] contact or contactlist by category shows errorpage when SEO is on

### ISSUE THREAD

**Issue by det13** [joomla/joomla-cms#32279]
### Steps to reproduce the issue
create a link to a contact or the contactlist by catagory in an article

### Expected result
the details of the contact(s) are shown when the link is clicked

### Actual result
an error page comes up with 'contact not found' or 'category not found'

### System information (as much as possible)
joomla 4.0 beta7 and 4.0 beta6, php 7.48

### Additional comments
when SEO is set to off, the links to a single contact or to the contactlist by category work as expected

**Comment by drmenzelit** [comment]
I can confirm.
If a menu item for contacts is present, the link in the article works also with SEO on.

**Comment by PhilETaylor** [comment]
Is this a linked to/duplicate of https://github.com/joomla/joomla-cms/issues/32248 ?

**Comment by richard67** [comment]
@PhilETaylor Maybe related, but I don't think duplicate, except if you can exactly reproduce for your issue what the author of this issue here mentioned in the "Additional comments" section of the description:

> when SEO is set to off, the links to a single contact or to the contactlist by category work as expected

**Comment by det13** [comment]
I can confirm, when there is a menu entry for example a list by category, then the link in an article is working too

**Comment by alikon** [comment]
#32279 vs #32248 related/linked  for sure :wink: 

maybe my old bad habit, 
I always add the item to a hidden menu  for SEO

**Comment by joomdonation** [comment]
Yes, the two issues are linked. The error happens with all of our components (even com_content), not just contact component. It happens when:

- No menu item created for the component.
- And **Remove IDs from URLs** config option in the component config set to **Yes**.

From what I see, it is hard to fix this issue with the current design of our routing system. Maybe we can implement a workaround: Only attach NomenuRules rule to component router if Remove IDs from URLs config option in the component config set to No. It is not perfect, but it is better than a 404 error like we are having at the moment.

**Comment by joomdonation** [comment]
@Hackwar Could you help looking at this issue?

**Comment by joomdonation** [comment]
@csthomas made a PR long time ago to fix this issue https://github.com/joomla/joomla-cms/pull/20979 . However, that PR is quite hard to follow.

**Comment by joomdonation** [comment]
This issue is the same with https://github.com/joomla/joomla-cms/issues/32248, so one of the two issues should be closed.

**Comment by richard67** [comment]
Closing as duplicate report of #32248 .<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32279">issues.joomla.org/tracker/joomla-cms/32279</a>.</sub>
